waspc/src/Wasp/Generator/DbGenerator/Common.hs

```diff
@@ -60,3 +66,42 @@ dbSchemaChecksumOnLastGenerateFileInDbRootDir = [relfile|schema.prisma.wasp-gene
 
 dbSchemaChecksumOnLastGenerateFileProjectRootDir :: Path' (Rel ProjectRootDir) (File DbSchemaChecksumOnLastGenerateFile)
 dbSchemaChecksumOnLastGenerateFileProjectRootDir = dbRootDirInProjectRootDir </> dbSchemaChecksumOnLastGenerateFileInDbRootDir
+
+data MigrateArgs = MigrateArgs
+  { _migrationName :: Maybe String,
+    _isCreateOnlyMigration :: Bool
+  }
+  deriving (Show, Eq)
+
+emptyMigrateArgs :: MigrateArgs
+emptyMigrateArgs = MigrateArgs {_migrationName = Nothing, _isCreateOnlyMigration = False}
+
+asArgs :: MigrateArgs -> [String]
+asArgs migrateArgs = do
+  concat . concat $
+    [ [["--create-only"] | _isCreateOnlyMigration migrateArgs],
+      [["--name", name] | Just name <- [_migrationName migrateArgs]]
+    ]
+
+parseMigrateArgs :: Maybe [String] -> MigrateArgs
+parseMigrateArgs Nothing = emptyMigrateArgs
+parseMigrateArgs (Just migrateArgs) = do
+  go migrateArgs emptyMigrateArgs
+  where
+    go :: [String] -> MigrateArgs -> MigrateArgs
+    go ("--create-only" : rest) mArgs = go rest mArgs {_isCreateOnlyMigration = True}
+    go ("--name" : name : rest) mArgs = go rest mArgs {_migrationName = Just name}
+    go _ mArgs = mArgs
+
+data OnLastDbConcurrenceChecksumAction
+  = WriteOnLastDbConcurrenceChecksum
+  | RemoveOnLastDbConcurrenceChecksum
+  | IgnoreOnLastDbConcurrenceChecksum
+
+-- When we do a create-only migration, we need to remove the DbSchemaChecksumOnLastDbConcurrenceFile so they know to migrate.
+-- When we migrate, we need to write it to indicate the local code and DB are in sync.
```

Comment:
In comments you are describing imlementation. Instead, would be better to define what we expect the function to do. I would go in the direction of explaining that this function, based on the args given, decides what we should do with the checksum file. Although, this should actually be visible from the type signature. So maybe we can improve the names there and avoid the comment all together!

Reply:
Good point, I can make it more about what we are doing at a higher level, and possibly remove it altogether if the sig can be cleaned up some to convey it! 👍🏻 It tends to be my crutch for, "this feels a bit complicated, just comment it to death", but that usually means the code itself can be improved to avoid it instead hehe

Reply:
The names are getting longer :D but more consistent now. I did feel the comment inside does help the reader some. Do you think this is any better?

```haskell
getOnLastDbConcurrenceChecksumFileRefreshAction :: MigrateArgs -> RefreshOnLastDbConcurrenceChecksumFile
getOnLastDbConcurrenceChecksumFileRefreshAction migrateArgs =
  -- Since a create-only migration allows users to write any SQL, we remove the file to force
  -- revalidation with the DB. If it is a regular migration, we write it since they will be in sync.
  if _isCreateOnlyMigration migrateArgs
    then RemoveOnLastDbConcurrenceChecksumFile
    else WriteOnLastDbConcurrenceChecksumFile
```

Reply:
I think this looks good!